src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -678,9 +678,13 @@ public override void VisitParameter(IParameterSymbol symbol)
             // (e.g. field types, param types, etc), which just want the name whereas parameters are
             // used on their own or in the context of methods.
 
-            var includeType = format.ParameterOptions.IncludesOption(SymbolDisplayParameterOptions.IncludeType);
+            // SignatureOnlyParameterSymbol.ContainingSymbol throws.
+            var isFunctionPointerParameter = symbol is not Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.Symbol { UnderlyingSymbol: SignatureOnlyParameterSymbol }
+                && symbol.ContainingSymbol is IMethodSymbol { MethodKind: MethodKind.FunctionPointerSignature };
+            var includeType = format.ParameterOptions.IncludesOption(SymbolDisplayParameterOptions.IncludeType)
+                || isFunctionPointerParameter;
```

Comment:
>|| isFunctionPointerParameter [](start = 16, length = 29)

I am not sure whether adding this special case here is the right thing to do. Perhaps once we visit a function pointer type we can simply adjust the format?

Reply:
I'm not sure this approach will work well, as it seems SymbolDisplayVisitor.VisitParameter can be called on a function pointer parameter in other contexts, and we'll need to at the very least fix the spacing in such a case.

For example it's called [here](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/SignatureHelp/InvocationExpressionSignatureHelpProviderBase_DelegateAndFunctionPointerInvoke.cs,102).

Perhaps the neatest solution to this is to expose on `IParameter` that it is a function pointer parameter

Reply:
> I'm not sure this approach will work well, as it seems SymbolDisplayVisitor.VisitParameter can be called on a function pointer parameter in other contexts, and we'll need to at the very least fix the spacing in such a case.

It feels like we don't need to make any changes in ```VisitParameter``` since we are handling everything in ```VisitMethod```. I do not feel any sympathy to callers that display parameters on their own. 

---
In reply to: [579027094](https://github.com/dotnet/roslyn/pull/51223#discussion_r579027094) [](ancestors = 579027094)

Reply:
> It feels like we don't need to make any changes in VisitParameter since we are handling everything in VisitMethod. I do not feel any sympathy to callers that display parameters on their own.

I've been trying to find a workaround to that case. Other than reimplementing all the logic to visit a FunctionParameter at the call site (which seems wrong), or exposing a new public method `SymbolDisplay.FunctionParameterToDisplayParts` I can't see anything better than handling a FunctionPointerParameterSymbol correctly in `VisitParameter`.

I'll go with adding `SymbolDisplay.FunctionParameterToDisplayParts` for now...

Reply:
> I'll go with adding SymbolDisplay.FunctionParameterToDisplayParts for now...

I'm not happy with that either. I think the correct solution is to have an IParameterSymbol publicly distinguishable from a FunctionParameterSymbol.

Reply:
> I'm not happy with that either. I think the correct solution is to have an IParameterSymbol publicly distinguishable from a FunctionParameterSymbol.

I suggest to leave this issue alone for now. The goal of this PR is to address issues with displaying function pointer type symbols as a whole. No one complained about display for parameter symbols within them and I am not even sure if there is a compelling scenario to display it on its own, we might not even care much about scenarios like that, apart from getting to perfection. Technically there is no method or even parameter behind the function pointer type, just a signature. A fake method symbol is just a way to capture information about the signature. I think we can fix the problem at hand without making any changes in this function. Again, I suggest let's focus on that. For any other outstanding issues that you think should be looked at, let's open separate issues with compelling scenarios (not just I call this API and I get this, looks strange), perhaps with solutions that you propose and let's wait for them to be triaged/evaluated by the team.

---
In reply to: [591115519](https://github.com/dotnet/roslyn/pull/51223#discussion_r591115519) [](ancestors = 591115519)

Reply:
I'm looking at this for the tests which are broken by this PR - signature help for calling a function pointer is incorrectly formatted

Reply:
> I'm looking at this for the tests which are broken by this PR - signature help for calling a function pointer is incorrectly formatted

Is it broken when the change in this function is reverted? Please provide more details how exactly it is broken and what change in this PR you think breaks it. Without this information it is pretty much impossible to provide any guidance.

---
In reply to: [591629245](https://github.com/dotnet/roslyn/pull/51223#discussion_r591629245) [](ancestors = 591629245)

Reply:
> this function 

To clarify, I am talking about ```VisitParameter```

---
In reply to: [591633037](https://github.com/dotnet/roslyn/pull/51223#discussion_r591633037) [](ancestors = 591633037,591629245)

Reply:
Yes